src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -696,7 +696,7 @@ protected override void Dispose(bool disposing)
                 }
                 finally
                 {
-                    _stream = null;
+                    _stream = null!;
```

Comment:
This shouldn't be necessary once you make _stream nullable.

Reply:
by the reason mentioned above i don't think _stream should be nullable, and its ok to set null here as it is Dispose() method

Reply:
It should be marked as nullable, that is the purpose of the feature, avoid null reference exceptions, if someone touches this code later on, should get the right warnings if they don't check for null.

Reply:
Agree about the purpose of the feature, but by design this field is would never be null unless the warpped stream is disposed, i don't think anyone who touch it later would set it to null except in dispose method, for me we don't need to add extra null check overhead such case.

Reply:
After thinking about it more I agree its better to be nullable, will update thanks